### PR TITLE
Added tests for scparams/variables list / BZ 1303653, 1219934

### PR DESCRIPTION
**Pull request by svtkachenko**
```python
% pytest -n 2 -v tests/foreman/cli/test_{environment,host,hostgroup}.py -k 'sc_params or puppet or list_s' 
===================================================================== test session starts =====================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.4, py-1.4.31, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile: 
plugins: xdist-1.15.0, html-1.12.0, cov-2.4.0
[gw0] linux2 Python 2.7.12 cwd: /home/qui/code/robottelo
[gw1] linux2 Python 2.7.12 cwd: /home/qui/code/robottelo
[gw0] Python 2.7.12 (default, Nov  7 2016, 11:55:55)  -- [GCC 6.2.1 20160830]
[gw1] Python 2.7.12 (default, Nov  7 2016, 11:55:55)  -- [GCC 6.2.1 20160830]
gw0 [16] / gw1 [16]
scheduling tests via LoadScheduling

tests/foreman/cli/test_environment.py::EnvironmentTestCase::test_positive_sc_params_by_id <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_environment.py::EnvironmentTestCase::test_positive_sc_params_by_name <- robottelo/decorators/__init__.py 
[gw0] PASSED tests/foreman/cli/test_environment.py::EnvironmentTestCase::test_positive_sc_params_by_id <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_puppet_class_id 
[gw1] PASSED tests/foreman/cli/test_environment.py::EnvironmentTestCase::test_positive_sc_params_by_name <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_puppet_class_name 
[gw0] PASSED tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_puppet_class_id 
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_scparams_by_id <- robottelo/decorators/__init__.py 
[gw1] PASSED tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_create_with_puppet_class_name 
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_smartvariables_by_id <- robottelo/decorators/__init__.py 
[gw0] PASSED tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_scparams_by_id <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_scparams_by_name <- robottelo/decorators/__init__.py 
[gw1] PASSED tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_smartvariables_by_id <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_smartvariables_by_name <- robottelo/decorators/__init__.py 
[gw0] PASSED tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_scparams_by_name <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_ca_proxy <- robottelo/decorators/__init__.py 
[gw1] PASSED tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_list_smartvariables_by_name <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_class_id 
[gw0] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_ca_proxy <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_class_name 
[gw0] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_class_name 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_scparams_by_id 
[gw1] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_class_id 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_proxy <- robottelo/decorators/__init__.py 
[gw1] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_puppet_proxy <- robottelo/decorators/__init__.py 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_smartvariables_by_id 
[gw0] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_scparams_by_id 
[gw1] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_smartvariables_by_id 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_scparams_by_name 
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_smartvariables_by_name 
[gw0] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_scparams_by_name 
[gw1] PASSED tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_smartvariables_by_name 

================================================================= 16 passed in 922.63 seconds =================================================================

pytest -v tests/foreman/cli/test_{classparameters,variables}.py -k 'list_by'
===================================================================== test session starts =====================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.4, py-1.4.31, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile: 
plugins: xdist-1.15.0, html-1.12.0, cov-2.4.0
collected 95 items 

tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_import_twice_list_by_puppetclass_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_env_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_env_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_host_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_host_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_hostgroup_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_hostgroup_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_puppetclass_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_by_puppetclass_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_variables.py::SmartVariablesTestCase::test_positive_list_by_host_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_variables.py::SmartVariablesTestCase::test_positive_list_by_host_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_variables.py::SmartVariablesTestCase::test_positive_list_by_hostgroup_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_variables.py::SmartVariablesTestCase::test_positive_list_by_hostgroup_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_variables.py::SmartVariablesTestCase::test_positive_list_by_puppetclass_id <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_variables.py::SmartVariablesTestCase::test_positive_list_by_puppetclass_name <- robottelo/decorators/__init__.py PASSED

===================================================================== 80 tests deselected =====================================================================
========================================================= 15 passed, 80 deselected in 825.29 seconds ==========================================================
```

### CODE REVIEW

File: tests/foreman/cli/test_hostgroup.py [hunk 1]
```diff
@@ -187,6 +211,38 @@ def test_positive_create_with_puppet_proxy(self):
             hostgroup['puppet-master-proxy-id'],
         )
 
+    @tier1
+    def test_positive_create_with_puppet_class_id(self):
+        """Check if hostgroup with puppet class can be created
```
Comment (ldjebran) [inline tests/foreman/cli/test_hostgroup.py]: Check if hostgroup with puppet class id can be created

File: tests/foreman/cli/test_hostgroup.py [hunk 2]
```diff
@@ -187,6 +211,38 @@ def test_positive_create_with_puppet_proxy(self):
...
+            'puppet-class-ids': self.puppet_class['id'],
+            'environment-id': self.env['id'],
+            'content-view-id': self.cv['id'],
+            'query-organization-id': self.org['id'],
+        })
+        self.assertIn(self.puppet_class['name'], hostgroup['puppetclasses'])
+
+    @tier1
+    def test_positive_create_with_puppet_class_name(self):
+        """Check if hostgroup with puppet class can be created
```
Comment (ldjebran) [inline tests/foreman/cli/test_hostgroup.py]: Check if hostgroup with puppet class name can be created

File: tests/foreman/cli/test_hostgroup.py [hunk 3]
```diff
@@ -501,3 +557,109 @@ def test_negative_delete_by_id(self):
...
+        })
+        # Override one of the sc-params from puppet class
+        sc_params_list = SmartClassParameter.list({
+            'environment': self.env['name'],
+            'search': u'puppetclass="{0}"'.format(self.puppet_class['name'])
+        })
+        scp_id = choice(sc_params_list)['id']
+        SmartClassParameter.update({'id': scp_id, 'override': 1})
+        # Verify that affected sc-param is listed
+        hg_scparams = HostGroup.sc_params({'hostgroup': hostgroup['name']})
```
Comment (ldjebran) [inline tests/foreman/cli/test_hostgroup.py]: the test doc say List all smart class parameters using hostgroup id

**Comment by svtkachenko** [comment]
Docstrings updated.

**Comment by svtkachenko** [comment]
```python
% py.test -v tests/foreman/cli/test_hostgroup.py -k 'test_positive_list_scparams_by_id'
===================================================================== test session starts =====================================================================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile: 
plugins: xdist-1.15.0, html-1.12.0, cov-2.3.1
collected 27 items 

tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_list_scparams_by_id PASSED

================================================ 26 tests deselected by '-ktest_positive_list_scparams_by_id' =================================================
========================================================== 1 passed, 26 deselected in 100.09 seconds ==========================================================
```

**Pull request merged**